Update bazel test job to 0.4.2 and support //build-tools/... and //build/...

ixdy opened the pull request:
Necessary for https://github.com/kubernetes/kubernetes/pull/38736 to work.

x-ref https://github.com/kubernetes/kubernetes/issues/38126

k8s-reviewable commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.kubernetes.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.kubernetes.io/reviews/kubernetes/test-infra/1373)
<!-- Reviewable:end -->

ixdy commented:
I'll rebase and fold the bazel update into this PR too.

ixdy commented:
I've rebased `jobs/pull-kubernetes-bazel.sh` and I've updated the bazel image to bazel 0.4.2.

I haven't pushed any images or configs yet.

I don't know how to test this, either.

spxtr commented on images/pull-kubernetes-bazel/Dockerfile:
```diff
@@ -29,9 +29,10 @@ RUN apt-get update && apt-get install -y \
-    dpkg -i bazel_0.3.2-linux-x86_64.deb && \
-    rm bazel_0.3.2-linux-x86_64.deb
+ENV BAZEL_VERSION 0.4.2
+RUN wget "https://github.com/bazelbuild/bazel/releases/download/${BAZEL_VERSION}/bazel_${BAZEL_VERSION}-linux-x86_64.deb" && \
```
Does it make sense to do `RUN BAZEL_VERSION=0.4.2 wget ...` instead?

spxtr commented on jobs/pull-kubernetes-bazel.sh:
```diff
@@ -28,8 +28,13 @@ release-1.0|release-1.1|release-1.2|release-1.3|release-1.4|release-1.5)
   ;;
 esac
 
+build_dir=build
```
TODO: remove this block?

ixdy replied:
done

spxtr commented:
I don't think we have a way to test job changes in other repos as a presubmit to this repo. That's certainly something we should add, though.

spxtr commented:
/lgtm

be sure to push the new image before updating the job config :)

ixdy commented:
Yep, `bazelbuild:0.1` has been pushed.
